### ISSUE TITLE
After upgrading from 3.0.3 to 3.3.8, the 'Flutter/Flutter.h' file was not found

### ISSUE THREAD

**Issue by HSCOO** [flutter/flutter#115619]
What I am doing now is an iOS project, I have connected to flutter, and it was running normally before 3.0.3. Because I used `Matrix4.setEntry` when I was doing animation, there was a freeze after a period of continuous operation. I found an answer in issues that required me to upgrade to above 3.3.0. But after I upgraded to 3.3.8, `the 'Flutter/Flutter.h' file was not found` appeared.

**flutter doctor**：

```shell
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.8, on macOS 12.6 21G115 darwin-x64, locale
    zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2021.2.4)
[✓] VS Code (version 1.71.2)
[✓] Connected device (2 available)
[✓] HTTP Host Availability
• No issues found!
```

After I upgraded to 3.3.8, I found that `Pods/Development pods/Flutter` no longer has a `Frameworks` directory. I found a useful information in **Flutter.podspec**:

```ruby
Pod::Spec.new do |s|
    s.name             = 'Flutter'
    s.version          = '1.0.0'
    s.summary          = 'A UI toolkit for beautiful and fast apps.'
    s.homepage         = 'https://flutter.dev'
    s.license          = { :type => 'BSD' }
    s.author           = { 'Flutter Dev Team' => 'flutter-dev@googlegroups.com' }
    s.source           = { :git => 'https://github.com/flutter/engine', :tag => s.version.to_s }
    s.ios.deployment_target = '11.0'
    # Framework linking is handled by Flutter tooling, not CocoaPods.
    # Add a placeholder to satisfy `s.dependency 'Flutter'` plugin podspecs.
    s.vendored_frameworks = 'path/to/nothing'
end

```

I still can't understand: `Framework linking is handled by Flutter tooling, not CocoaPods`. Do I need to execute any commands? Or just compile the main project. But at present, this problem will appear as long as the project is compiled: `the 'Flutter/Flutter.h' file was not found`.

I also noticed that there is a description of `Flutter.framework` in `podhelper.rb`, but I don't know if it helps me.

```ruby
def install_flutter_application_pod(flutter_application_path)
  flutter_application_path ||= File.join('..', '..')

  export_script_directory = File.join(flutter_application_path, '.ios', 'Flutter')

  # Keep script phase paths relative so they can be checked into source control.
  relative = flutter_relative_path_from_podfile(export_script_directory)

  flutter_export_environment_path = File.join('${SRCROOT}', relative, 'flutter_export_environment.sh');

  # Compile App.framework and move it and Flutter.framework to "BUILT_PRODUCTS_DIR"
  script_phase :name => 'Run Flutter Build vbf_module Script',
    :script => "set -e\nset -u\nsource \"#{flutter_export_environment_path}\"\nexport VERBOSE_SCRIPT_LOGGING=1 && \"$FLUTTER_ROOT\"/packages/flutter_tools/bin/xcode_backend.sh build",
    :execution_position => :before_compile

  # Embed App.framework AND Flutter.framework.
  script_phase :name => 'Embed Flutter Build vbf_module Script',
    :script => "set -e\nset -u\nsource \"#{flutter_export_environment_path}\"\nexport VERBOSE_SCRIPT_LOGGING=1 && \"$FLUTTER_ROOT\"/packages/flutter_tools/bin/xcode_backend.sh embed_and_thin",
    :execution_position => :after_compile
end
```

I don't know if I understand correctly, the timing of `Flutter.framework` embedding is too late for me. At least the `Embed Flutter Build vbf_module Script` has not been executed before compiling the error.

**Comment by darshankawar** [comment]
@HSCOO 
Can you provide `flutter run --verbose` that shows the reported error log ? It will be huge so you can put that in a text file and attach it here.

Also, if possible, provide a minimal code sample or a sample repo that we can clone which triggers the error you are seeing.

There were similar issues reported as below with workarounds mostly for your reference:

https://github.com/flutter/flutter/issues/96063#issuecomment-1004656518
https://github.com/flutter/flutter/issues/88634#issuecomment-908939116
https://github.com/flutter/flutter/issues/42956#issuecomment-549623011

**Comment by HSCOO** [comment]
@darshankawar  I'm very sorry, but the three issues you provided didn't help me at all. After I upgraded to 3.3.8, I created a brand new demo without any modification, which can be run. This is enough to show that my environment can work.
I need to provide `flutter run --verbose` later. I want to know where did 3.3.8 put `Flutter.framework`? I have found evidence that cocoapod cannot manage `Flutter.framework`. And the meaning of `Framework linking is handled by Flutter tooling, not CocoaPods`.

**Comment by darshankawar** [comment]
> three issues you provided didn't help me at all.

That's ok. It was just for your reference. 

Can you provide the output of your `/users/<username>/Developement/SDKs/flutter/bin/cache/artifacts/engine/ios/Flutter.xcframework/` and check if any of the archs mentioned at that path contains `Flutter.framework/Flutter` ?

Also, since you already upgraded to latest, but can you provide output of `ls -R bin/cache/artifacts/engine/ios` ?

**Comment by HSCOO** [comment]
@darshankawar   The output of `ls -R bin/cache/artifacts/engine/ios` is: 
```shell
Flutter.podspec     LICENSE             gen_snapshot_armv7
Flutter.xcframework gen_snapshot_arm64

bin/cache/artifacts/engine/ios/Flutter.xcframework:
Info.plist                 ios-arm64_x86_64-simulator
ios-arm64_armv7

bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_armv7:
Flutter.framework

bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_armv7/Flutter.framework:
Flutter        Info.plist     _CodeSignature
Headers        Modules        icudtl.dat

bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_armv7/Flutter.framework/Headers:
Flutter.h                           FlutterEngineGroup.h
FlutterAppDelegate.h                FlutterHeadlessDartRunner.h
FlutterBinaryMessenger.h            FlutterMacros.h
FlutterCallbackCache.h              FlutterPlatformViews.h
FlutterChannels.h                   FlutterPlugin.h
FlutterCodecs.h                     FlutterPluginAppLifeCycleDelegate.h
FlutterDartProject.h                FlutterTexture.h
FlutterEngine.h                     FlutterViewController.h

bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_armv7/Flutter.framework/Modules:
module.modulemap

bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_armv7/Flutter.framework/_CodeSignature:
CodeResources

bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_x86_64-simulator:
Flutter.framework

bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_x86_64-simulator/Flutter.framework:
Flutter        Info.plist     _CodeSignature
Headers        Modules        icudtl.dat

bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_x86_64-simulator/Flutter.framework/Headers:
Flutter.h                           FlutterEngineGroup.h
FlutterAppDelegate.h                FlutterHeadlessDartRunner.h
FlutterBinaryMessenger.h            FlutterMacros.h
FlutterCallbackCache.h              FlutterPlatformViews.h
FlutterChannels.h                   FlutterPlugin.h
FlutterCodecs.h                     FlutterPluginAppLifeCycleDelegate.h
FlutterDartProject.h                FlutterTexture.h
FlutterEngine.h                     FlutterViewController.h

bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_x86_64-simulator/Flutter.framework/Modules:
module.modulemap

bin/cache/artifacts/engine/ios/Flutter.xcframework/ios-arm64_x86_64-simulator/Flutter.framework/_CodeSignature:
CodeResources

```

I observed that the `/users/<username>/Developement/SDKs/flutter/bin/cache/artifacts/engine/ios/Flutter.podspec` is different from the file `my_flutter_module/.ios/Flutter/engine/Flutter.podspec` I generated.

The `/users/<username>/Developement/SDKs/flutter/bin/cache/artifacts/engine/ios/Flutter.podspec` like this:
```ruby

Pod::Spec.new do |s|
  s.name             = 'Flutter'
  s.version          = '1.0.0'
  s.summary          = 'High-performance, high-fidelity mobile apps.'
  s.description      = <<-DESC
Flutter provides an easy and productive way to build and deploy high-performance mobile apps for Android and iOS.
                       DESC
  s.homepage         = 'https://flutter.io'
  s.license          = { :type => 'MIT' }
  s.author           = { 'Flutter Dev Team' => 'flutter-dev@googlegroups.com' }
  s.source           = { :git => 'https://github.com/flutter/engine', :tag => s.version.to_s }
  s.ios.deployment_target = '8.0'
  s.vendored_frameworks = 'Flutter.framework'
end

```

The `my_flutter_module/.ios/Flutter/engine/Flutter.podspec` like this:
```ruby
Pod::Spec.new do |s|
  s.name             = 'Flutter'
  s.version          = '1.0.0'
  s.summary          = 'A UI toolkit for beautiful and fast apps.'
  s.description      = <<-DESC
Flutter is Google's UI toolkit for building beautiful, fast apps for mobile, web, desktop, and embedded devices from a single codebase.
This pod vends the iOS Flutter engine framework. It is compatible with application frameworks created with this version of the engine and tools.
                       DESC
  s.homepage         = 'https://flutter.dev'
  s.license          = 'BSD'
  s.author           = { 'Flutter Dev Team' => 'flutter-dev@googlegroups.com' }
  s.source           = { :git => 'https://github.com/flutter/engine', :tag => s.version.to_s }
  s.documentation_url     = 'https://flutter.dev/docs'
  s.ios.deployment_target = '9.0'
  s.vendored_frameworks = 'Flutter.xcframework'
end
```

**Comment by darshankawar** [comment]
Thanks for the update. Take a look at [this](https://docs.flutter.dev/development/add-to-app/ios/project-setup) document for detailed explaination of how ios module works with Flutter and see if it helps in your case.

**Comment by HSCOO** [comment]
@darshankawar My question is not how to integrate flutter to iOS project, I have already integrated it. And I have iterated many versions, currently using version **3.0.3.** Only after upgrading to **3.3.8** will the problem of not being able to find Flutter.framework appear. And it's not that I can't find `Flutter.framework `in my Flutter SDK, but that I can't find `Flutter.framework` in the iOS project after running the `podhelper.rb script`. The [current tutorial](https://docs.flutter.dev/development/add-to-app/ios/project-setup) says `The podhelper.rb script embeds your plugins, Flutter.framework, and App.framework into your project.`, but it is not found in my project.

**Comment by darshankawar** [comment]
Thanks for the update @HSCOO 
Can you try by switching to latest master first and see if on that version too, you get same error ?

`flutter channel master`
`flutter upgrade`

**Comment by shingohu** [comment]
I have the same  question after upgrade 3.0.5 ->3.3.9
the podfile config

```
#flutter本地调试
def debugFlutterPods
  flutter_application_path = '../xxx'
  load File.join(flutter_application_path, '.ios', 'Flutter', 'podhelper.rb')
  install_all_flutter_pods(flutter_application_path)
end

target 'xxx' do
  debugFlutterPods
end

post_install do |installer|
  flutter_post_install(installer) if defined?(flutter_post_install)
  

```

The xcode build fail  with `No such module 'Flutter'`

**Comment by HSCOO** [comment]
I have found the root cause. The podhelp.rb used under version 3.3.8 finally comes from `fluttersdkpath/packages/flutter_tools/bin/podhelper.rb`. There is a code in the `flutter_additional_ios_build_settings` method that explains the answer:
```ruby
# Skip other updates if it's not a Flutter plugin (transitive dependency).
next unless target.dependencies.any? { |dependency| dependency.name == 'Flutter' }
```
Then you only need to manually add flutter dependencies in the module xxx.podspec that reported the error:
```ruby
s.dependency 'Flutter'
```

**Comment by HSCOO** [comment]
I've also been looking for Flutter.framework? The original new reference method directly references Flutter.framewrok under the flutersdk path.
```
fluttersdkpatch/bin/cache/artifacts/engine/ios/Flutter.xcframework
```

**Comment by darshankawar** [comment]
@HSCOO 
Since the original issue has been resolved, I'll go ahead and close this.
Regarding your above comment, you may open a new issue for better tracking.
